internal/services/appservice/helpers/service_plan.go

```diff
@@ -137,7 +137,7 @@ func PlanTypeFromSku(input string) string {
 }
 
 // ServicePlanInfoForApp returns the OS type and Service Plan SKU for a given App Service Resource
-func ServicePlanInfoForApp(ctx context.Context, metadata sdk.ResourceMetaData, id interface{}) (osType *string, planSku *string, err error) {
+func ServicePlanInfoForApp(ctx context.Context, metadata sdk.ResourceMetaData, id interface{}, serviceFarmId string) (osType *string, planSku *string, err error) {
```

Comment:
This function is intended to retrieve the Service Plan ID from the App ID, it should not take the ID directly to circumvent this.